ansible_runner/utils/__init__.py

```diff
@@ -464,3 +465,19 @@ def get_executable_path(name):
     if exec_path is None:
         raise ConfigurationError(f"{name} command not found")
     return exec_path
+
+
+def signal_handler():
+    # Only the main thread is allowed to set a new signal handler
+    if threading.current_thread() is not threading.main_thread():
+        return None
+
+    signal_event = threading.Event()
+    # closure to set signal event
+    def _handler(number, frame):
```

Comment:
we want to create the signal event
.. then pass it in to a closure
.. that will trigger the signal
.. and finally register the function handle / callable to `signal.signal`

When the associated signal happens `_handler` gets called, the signal event is registered, `signal_event.is_set` becomes true, meaning that `cancel_callback` becomes true and 🥁  ansible-runner knows it's time to clean up

Reply:
This looks great!